ravenframework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping_snapshot.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on Oct. 10, 2022
+
+  @author: Shiming Yin
+
+  This class is for the algorithms of Snapshot Physics-guided Coverage Mapping
+  It inherits from the PostProcessor directly
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import numpy as np
+import xarray as xr
+from scipy import stats
+# from sklearn.linear_model import OrthogonalMatchingPursuit
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from ravenframework.utils import InputData, InputTypes
+from .. import ValidationBase
+#Internal Modules End--------------------------------------------------------------------------------
+
+class PhysicsGuidedCoverageMapping_snapshot(ValidationBase):
```

Comment:
@ShimingYIN Please rework on this PR. The class "PhysicsGuidedCoverageMapping_snapshot" should inherit from "PhysicsGuidedCoverageMapping" or it should be combined with existing "PhysicsGuidedCoverageMapping" class. If  the updates mainly focus on the loop over time, I would suggest to integrate this class with "PhysicsGuidedCoverageMapping" instead to write a new class. Please let me know if you have any question.

Reply:
Understood. I'll modify this problem on class inheritance and pull the request again